src/Participants/Robot.java

```diff
@@ -11,33 +11,24 @@ public Robot(String name, double runDistance, double jumpHeight) {
         this.jumpHeight = jumpHeight;
     }
 
-    public void run(Racetrack obstacle) {
+    public boolean run(Racetrack obstacle) {
```

Comment:
again, you don't need obstacle object here. Distance is enough, and the code use/readability will dramatically increase